### ISSUE TITLE
assert_never(Never) after sys.exit(0) emits type-assertion-failure

### ISSUE THREAD

**Issue by eclbg** [astral-sh/ty#3082]
### Summary

Playground reproducer: https://play.ty.dev/44d1736f-8b77-4564-bb99-844ab546aa22

Same reproducer inline:
```python
import sys

from typing_extensions import assert_never, Never

def func():
    if True:
        pass
    else:
        assert_never(Never)  # No diagnostic here, as expected

sys.exit(0)
assert_never(Never)  # This emits an error diagnostic: type-assertion-failure
```

### Version

Confirmed in 0.0.19 and latest (playground)

I'd like to work on this myself if it's a good first (second, actually) issue. I have not yet investigated what the scope of this fix will be.

**Comment by sharkdp** [comment]
The reason for this is described here https://github.com/astral-sh/ty/issues/2480#issuecomment-3890737338

We do not analyze calls at the module level. If you do the same in a function scope, you get no diagnostic:
```py
def _():
    sys.exit(0)  # now this is detected as a terminal call
    assert_never(Never)
```

I actually have a draft PR open for this already (https://github.com/astral-sh/ruff/pull/23245), but I need to figure out what to do with the micro-benchmark regression.

**Comment by eclbg** [comment]
Thanks for the context :) And apologies for the dupe!
